src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpCodeVisitor.cs

```diff
@@ -116,6 +117,45 @@ protected override void Visit(TemplateChunk chunk)
             Writer.WriteEndMethodInvocation(false).WriteLine();
         }
 
+        protected override void Visit(ParentLiteralChunk chunk)
+        {
+            if (Context.Host.DesignTimeMode)
+            {
+                // Skip generating the chunk if we're in design time or if the chunk is empty.
+                return;
+            }
+            
+            if (Context.Host.EnableInstrumentation)
+            {
+                var start = chunk.Children[0].Association.Start.AbsoluteIndex;
+
+                var length = 0;
+                for (var i = 0; i < chunk.Children.Count; i++)
+                {
+                    length += chunk.Children[i].Association.Length;
```

Comment:
Can't we derive this from `chunk.GetText().Length`? You call `.GetText()` below anyways.